### ISSUE TITLE
Autocomplete for forms

### ISSUE THREAD

**Issue by nrfox** [kiali/kiali#6803]
Kiali has a number of forms, namely to create istio objects. These forms could be greatly enhanced with autocomplete. Autocomplete makes it much easier for users to use forms by narrowing the scope of options and working with partial data (you only remember part of the name of a field). As a good example of where autocomplete would be useful, take a look at the form to create an `AuthorizationPolicy`:

![Screenshot from 2023-10-30 14-55-42](https://github.com/kiali/kiali/assets/6226732/70ad4e62-c5ce-41e9-8400-6ae2b27a717e)

Going down the list of fields:

`Labels` could autocomplete with the workload labels that are present in the `travel-portal` namespace.

![Screenshot from 2023-10-30 14-56-01](https://github.com/kiali/kiali/assets/6226732/433b6129-c030-4553-9a03-a0254b05fa86)

`namespaces` could autocomplete with the list of namespaces.

`when.conditions` could autocomplete with the fixed [auth policy conditions fields](https://istio.io/latest/docs/reference/config/security/conditions/).

These are just a few examples from one form but most likely other forms are similar. In most or all of these cases, kiali already has all the information necessary to autocomplete the fields.

**Comment by jshaughn** [comment]
This would be nice, wonder if PatternFly has some support for this?

**Comment by fkargbo** [comment]
> This would be a nice enhancement, and could be done incrementally, starting with Wizards we think are the most critical.
> 
> @fkargbo, Do you know if PatternFly has anything around autocomplete, whether technical or convention?

@jshaughn Yes, PatternFly does support autocomplete. Here are some basic examples. They can get specific as per requirements.  
![image](https://github.com/kiali/kiali/assets/68336320/74eaba3b-d260-4a9a-b24d-307e7b5fb436)
![image](https://github.com/kiali/kiali/assets/68336320/60e0de09-d4f1-48a2-92c9-e524c4f5c452)

**Comment by jshaughn** [comment]
A nice enhancement but I don't think the wizard use merits the work. Perhaps if we attack a particular feature we can add it individually, but in that case we will use a dedicated issue. Closing this blanket issue.
